File: src/test/java/com/github/michael72/pumlsrv/UmlConverterTest.java
package com.github.michael72.pumlsrv;

import static org.junit.jupiter.api.Assertions.*;

import java.io.IOException;

import org.junit.jupiter.api.Test;

class UmlConverterTest {

  static String HELLO_BOB_ENC = "SyfFKj2rKt3CoKnELR1Io4ZDoSa70000";
  static String HELLO_BOB = "@startuml\n" + "Bob -> Alice : hello\n" + "@enduml";

  @Test
  void testEncodeUml() {
    try {
      final String enc = UmlConverter.encode(HELLO_BOB);
      assertEquals(HELLO_BOB_ENC, enc);
    } catch (IOException e) {
      fail("exception " + e.toString());
    }
  }

  @Test
  void testDecodeUml() {
    try {
      final String dec = UmlConverter.decode(HELLO_BOB_ENC);
      assertEquals(HELLO_BOB, dec);
    } catch (IOException e) {
      fail("exception " + e.toString());
    }
  }

}

File: src/test/kotlin/com/github/michael72/pumlsrv/UmlConverterTest.kt
package com.github.michael72.pumlsrv

import java.io.IOException
import org.junit.jupiter.api.Assertions.assertEquals
import org.junit.jupiter.api.Assertions.fail
import org.junit.jupiter.api.Test

class UmlConverterTest {

    companion object {
        const val HELLO_BOB_ENC = "SyfFKj2rKt3CoKnELR1Io4ZDoSa70000"
        const val HELLO_BOB = "@startuml\nBob -> Alice : hello\n@enduml"
    }

    @Test
    fun testEncodeUml() {
        try {
            val enc = UmlConverter.encode(HELLO_BOB)
            assertEquals(HELLO_BOB_ENC, enc)
        } catch (e: IOException) {
            fail("exception ${e}")
        }
    }

    @Test
    fun testDecodeUml() {
        try {
            val dec = UmlConverter.decode(HELLO_BOB_ENC)
            assertEquals(HELLO_BOB, dec)
        } catch (e: IOException) {
            fail("exception ${e}")
        }
    }
}
